File: SNS.Web/src/app/shared/contracts/community-snapshot.dto.ts
import { CommunityType } from './community-type';

/**
 * Represents a lightweight community snapshot for embedding across posts, comments, search, and consuming features.
 */
export interface CommunitySnapshotDto {
    id: string;
    name: string;
    type: CommunityType;
    logoUrl?: string | null;
}

File: backend/SNS.Application/ContentManagement/Communities/Communities/Contracts/CommunitySnapshotDto.cs
using SNS.Domain.ContentManagement.Communities.Enums;

namespace SNS.Application.ContentManagement.Communities.Communities.Contracts;

/// <summary>
/// Represents a lightweight snapshot of community information for embedding in posts, comments, search, and consuming features.
/// </summary>
/// <param name="Id">The unique identifier of the community.</param>
/// <param name="Name">The name of the community.</param>
/// <param name="Type">The privacy/visibility type of the community.</param>
/// <param name="LogoUrl">Optional resolved temporary logo URL.</param>
public sealed record CommunitySnapshotDto(
    Guid Id,
    string Name,
    CommunityType Type,
    string? LogoUrl
);
